File: Suspension matlab/spring/Unsprung_mass/Unsprung_mass.m
%% Unsprung mass
clear; clc; close all;
scriptInfo = dbstack('-completenames');
scriptDir = fileparts(scriptInfo(1).file);
addpath(scriptDir);
params = load_data(fullfile(scriptDir,'data.csv'));
paramNames = fieldnames(params);
for paramIdx = 1:numel(paramNames)
    eval([paramNames{paramIdx} ' = params.(paramNames{paramIdx});']);
end

%% Parameters

Ks = unsprung_Ks;    % spring stiffness (N/m)
MR = unsprung_MR;    % motion ratio
Kt = unsprung_Kt;    % tire stiffness (N/m)

%% Wheel rate

Kw = Ks * MR^2;

%% Unsprung mass range

mu = unsprung_mu;   % kg

%% Wheel hop frequency

fwh = (1/(2*pi)) * sqrt((Kw + Kt)./mu);

%% Plot

figure;
plot(mu,fwh,'LineWidth',2);

grid on;
xlabel('Unsprung Mass (kg)');
ylabel('Wheel Hop Frequency (Hz)');
title('Unsprung Mass vs Wheel Hop Frequency');

File: Suspension matlab/spring/Unsprung_mass/Unsprung_mass.py
from pathlib import Path
import sys

import matplotlib.pyplot as plt
import numpy as np

ROOT = next(p for p in Path(__file__).resolve().parents if (p / "suspension_utils.py").exists())
sys.path.insert(0, str(ROOT))

from suspension_utils import load_data, show_plots


def main() -> None:
    p = load_data(Path(__file__).with_name("data.csv"))
    ks = p["unsprung_Ks"]
    mr = p["unsprung_MR"]
    kt = p["unsprung_Kt"]
    mu = np.asarray(p["unsprung_mu"], dtype=float)

    kw = ks * mr**2
    fwh = (1 / (2 * np.pi)) * np.sqrt((kw + kt) / mu)

    plt.figure()
    plt.plot(mu, fwh, linewidth=2)
    plt.grid(True)
    plt.xlabel("Unsprung Mass (kg)")
    plt.ylabel("Wheel Hop Frequency (Hz)")
    plt.title("Unsprung Mass vs Wheel Hop Frequency")
    show_plots()


if __name__ == "__main__":
    main()
